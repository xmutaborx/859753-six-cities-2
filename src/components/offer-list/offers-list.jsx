import React from 'react';
import PropTypes from 'prop-types';

import OfferCard from '../offer-card/offer-card.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';

const OffersList = (props) => {
  const {onChangeActiveItem, onClearItem, offers, nearMode} = props;
  let wrappedClasses = `cities__places-list places__list tabs__content`;

  if (nearMode) {
    wrappedClasses = `near-places__list places__lis`;
  }

  return (
    <div className={wrappedClasses} >
      {offers.map((card) =>
        <OfferCard
          key={card.id}
          id={card.id}
          isFavorite={card.is_favorite}
          title={card.title}
          price={card.price}
          type={card.type}
          isPremium={card.is_premium}
          image={card.preview_image}
          rating={card.rating}
          nearMode={nearMode}
          onChangeActiveItem={() => onChangeActiveItem(card)}
          onClearItem={onClearItem}
        />
      )}
    </div>
  );
};

OffersList.defaultProps = {
  nearMode: false,
};

OffersList.propTypes = {
  onChangeActiveItem: PropTypes.func.isRequired,
  onClearItem: PropTypes.func.isRequired,
  offers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    // eslint-disable-next-line camelcase
    is_premium: PropTypes.bool.isRequired,
    // eslint-disable-next-line camelcase
    is_favorite: PropTypes.bool.isRequired,
    images: PropTypes.array.isRequired,
    rating: PropTypes.number.isRequired,
  })),
  nearMode: PropTypes.bool,
};

export {OffersList};
export default withActiveItem(OffersList);
