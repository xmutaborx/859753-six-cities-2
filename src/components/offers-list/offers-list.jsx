import React from 'react';
import PropTypes from 'prop-types';
import {PROP_TYPES_OFFERS_LIST} from '../../types/types';

import OfferCard from '../offer-card/offer-card.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';
import FavoritesCard from '../favorites-card/favorites-card.jsx';

const OffersList = (props) => {
  const {onChangeActiveItem, onClearItem, offers, nearMode, favoritesMode, favoritesCity} = props;
  let wrappedClasses = `cities__places-list places__list tabs__content`;

  if (nearMode) {
    wrappedClasses = `near-places__list places__lis`;
  }

  if (favoritesMode) {
    return (
      <ul className="favorites__list">
        {favoritesCity.map((city) => (
          <li className="favorites__locations-items" key={city}>
            <div className="favorites__locations locations locations--current">
              <div className="locations__item">
                <a className="locations__item-link" href="#">
                  <span>{city}</span>
                </a>
              </div>
            </div>
            <div className="favorites__places">
              {offers.map((offer) => {
                if (offer.city.name === city) {
                  return <FavoritesCard
                    card={offer}
                    isFavorite={offer.is_favorite}
                    title={offer.title}
                    key={offer.id}
                    price={offer.price}
                    type={offer.type}
                    image={offer.preview_image}
                    rating={offer.rating}
                    id={offer.id}
                  />;
                } else {
                  return null;
                }
              })}
            </div>
          </li>
        ))}
      </ul>
    );
  }

  return (
    <div className={wrappedClasses} >
      {offers.map((card) =>
        <OfferCard
          key={card.id}
          id={card.id}
          isFavorite={card.is_favorite}
          title={card.title}
          price={card.price}
          type={card.type}
          isPremium={card.is_premium}
          image={card.preview_image}
          rating={card.rating}
          nearMode={nearMode}
          onChangeActiveItem={() => onChangeActiveItem(card)}
          onClearItem={onClearItem}
        />
      )}
    </div>
  );
};

OffersList.defaultProps = {
  nearMode: false,
  favoritesMode: false,
};

OffersList.propTypes = {
  onChangeActiveItem: PropTypes.func.isRequired,
  onClearItem: PropTypes.func.isRequired,
  offers: PROP_TYPES_OFFERS_LIST,
  nearMode: PropTypes.bool,
  favoritesMode: PropTypes.bool,
  favoritesCity: PropTypes.arrayOf(PropTypes.string)
};

export {OffersList};
export default withActiveItem(OffersList);
