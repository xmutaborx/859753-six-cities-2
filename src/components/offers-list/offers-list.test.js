import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";

import {OffersList} from './offers-list.jsx';
import OFFER_MOCK from '../../mocks/offer-mock';

Enzyme.configure({adapter: new Adapter()});

it(`Offers Page correctly renders`, () => {
  const tree = shallow(<OffersList
    onChangeActiveItem={() => {}}
    onClearItem={() => {}}
    offers={OFFER_MOCK}
    favoritesCity={[`Amsterdam`]}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
