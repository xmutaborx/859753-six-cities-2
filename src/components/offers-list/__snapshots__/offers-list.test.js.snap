// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers Page correctly renders 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <Connect(OfferCard)
    id={1}
    image="img/1.png"
    isFavorite={false}
    isPremium={false}
    key="1"
    nearMode={false}
    onChangeActiveItem={[Function]}
    onClearItem={[Function]}
    price={120}
    rating={4.8}
    title="Beautiful & luxurious studio at great location"
    type="apartment"
  />
  <Connect(OfferCard)
    id={2}
    image="img/1.png"
    isFavorite={false}
    isPremium={false}
    key="2"
    nearMode={false}
    onChangeActiveItem={[Function]}
    onClearItem={[Function]}
    price={400}
    rating={5}
    title="Beautiful & luxurious studio at great location"
    type="apartment"
  />
</div>
`;
